easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -63,9 +63,19 @@ def yaml_join(loader, node):
     seq = loader.construct_sequence(node)
     return ''.join([str(i) for i in seq])
 
-# register the tag handler
-yaml.add_constructor('!join', yaml_join)
+def yaml_versionmajorminor(loader, node):
+    """
+    defines custom YAML function to get major version.
+    @param loader: the YAML Loader
+    @param node: the YAML (scalar) node
+    """
+    pyver = loader.construct_sequence(node)
```

Comment:
add this:

```python
if isinstance(pyver, list):
    pyver = pyver[0]
```

and maybe add a debug log message too in that case?

anyway, this shouldn't be needed, there must be a better way